block-storage/src/test/scala/coop/rchain/blockstorage/deploy/LMDBDeployStorageTest.scala

```diff
@@ -0,0 +1,55 @@
+package coop.rchain.blockstorage.deploy
+
+import java.nio.file.Path
+
+import cats.effect.{Resource, Sync}
+import cats.syntax.functor._
+import coop.rchain.blockstorage.util.io.IOError.RaiseIOError
+import coop.rchain.blockstorage.util.io._
+import coop.rchain.catscontrib.TaskContrib.TaskOps
+import coop.rchain.models.blockImplicits.signedDeployDataGen
+import monix.eval.Task
+import monix.execution.Scheduler
+import org.scalacheck.Gen
+import org.scalatest.{BeforeAndAfterAll, FlatSpecLike, Matchers}
+import org.scalatest.prop.GeneratorDrivenPropertyChecks
+
+class LMDBDeployStorageTest
+    extends FlatSpecLike
+    with Matchers
+    with GeneratorDrivenPropertyChecks
+    with BeforeAndAfterAll {
+
+  val scheduler = Scheduler.fixedPool("lmdb-deploy-storage-test-scheduler", 4)
+
+  implicit val raiseIOError: RaiseIOError[Task] = IOError.raiseIOErrorThroughSync[Task]
+
+  def withStorageLocation[R](f: Path => Task[R]): R = {
+    val testProgram = Sync[Task].bracket {
+      createTemporaryDirectory[Task]("lmdb-deploy-storage")
+    } { storagePath =>
+      f(storagePath)
+    } { storagePath =>
+      deleteRecursively[Task](storagePath).void
```

Comment:
Why not just delete the whole folder `storagePath`?

Reply:
Annoyingly, there is no easy way to delete the whole folder without getting [DirectoryNotEmptyException](https://docs.oracle.com/javase/10/docs/api/java/nio/file/DirectoryNotEmptyException.html) using Java's standard library.

Reply:
There are utilities in `apache-commons` - a minor convenience

Reply:
It would be great to have this information in a comment for easier understanding later.

Reply:
I don't think we have `apache-commons` in dependencies. The code for deleting a directory recursively only takes 7 lines so I couldn't justify bringing it in to just save us 7 lines of code.

I have added a comment as @tgrospic suggested.